src/installer/corehost/cli/deps_entry.cpp

```diff
@@ -99,6 +100,22 @@ bool deps_entry_t::to_path(const pal::string_t& base, const pal::string_t& ietf_
     else
     {
         trace::verbose(_X("    %s path query exists %s"), query_type, candidate.c_str());
+
+        // If a file is resolved to the servicing directory, disable it in the bundle directory.
+        // This step is necessary because runtime resolves assemblies in the bundle even before
+        // attempting to resolve using the TPA. Disabling the file's entry in the bundle
+        // ensures that the servicing entry in the TPA gets priority.
```

Comment:
@elinor-fung  I agree that the processing in the various `to_path` routines is a bit confusing. 
I've simplified the cases through which `is_servicing` flows a bit -- only through `to_full_path -> to_rel_path -> to_path`. 

A property of the current code is that the probing (whether bundle or file) is done only in `to_path`. 
I'd like to keep it this way, rather than have the caller again repeat the `sub_path` computations and probe/disable the file in bundle. 

I agree that this code can be refactored better, but I'd prefer to not do it in ask-mode. Thanks.

Reply:
Okay - can we at least update the description in the comments for the method (line 24 - I can't comment on the line in GitHub) and the header to call out that this function handles this disabling, so can affect broader state in the app now? (A `to_path` const function just sounds so benign relative to what it is actually responsible for now).